tests/test_export_live_query_protocol_projection.py

```
@@ -53,3 +59,335 @@ def test_projection_script_fails_when_minimum_query_count_not_met(tmp_path: Path
+
+    _assert_existing_artifacts_unchanged(previous)
+    _assert_no_publication_temporary_files(tmp_path)
+
+
+@pytest.mark.parametrize("failed_artifact", ["projection", "summary", "constraints"])
+def test_projection_replacement_failure_rolls_back_all_existing_artifacts(
+    tmp_path: Path, monkeypatch, failed_artifact: str
+) -> None:
+    paths = _projection_paths(tmp_path)
+    previous = _write_existing_artifacts(paths)
+    targets = dict(zip(("projection", "summary", "constraints"), paths))
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)